cli/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -250,14 +250,18 @@ impl Environment for ManagedEnvironment {
 
         let result = temporary.edit(flox, contents)?;
 
-        if matches!(result, EditResult::Success | EditResult::ReActivateRequired) {
-            generations
-                .add_generation(&mut temporary, "manually edited".to_string())
-                .map_err(ManagedEnvironmentError::CommitGeneration)?;
-            self.lock_pointer()?;
-            temporary.link(flox, &self.out_link)?;
+        if result == EditResult::Unchanged {
+            return Ok(result);
         }
 
+        dbg!("environment changed");
```

Comment:
Question: did you mean to use `debug!`?

Reply:
it was mostly debugging, but we can also add a debug! call when we're amending it anyway

```suggestion
        debug!("Environment changed, create generation, lock generation, build and link");
```